arm-streamanalytics/2016-03-01/swagger/functions.json

```diff
@@ -0,0 +1,768 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/StreamingJobNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FunctionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The test operation completed successfully.",
+            "schema": {
+              "$ref": "./inputs.json#/definitions/ResourceTestStatus"
+            }
+          },
+          "202": {
+            "description": "The test request was successfully initiated."
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.StreamAnalytics/streamingjobs/{jobName}/functions/{functionName}/RetrieveDefaultDefinition": {
+      "post": {
+        "tags": [
+          "Functions"
+        ],
+        "operationId": "Functions_RetrieveDefaultDefinition",
+        "description": "Retrieves the default definition of a function based on the parameters specified.",
+        "x-ms-examples": {
+          "Retrieve the default definition for an Azure ML function": { "$ref": "../examples/Function_RetrieveDefaultDefinition_AzureML.json" }
+        },
+        "parameters": [
+          {
+            "name": "functionRetrieveDefaultDefinitionParameters",
+            "in": "body",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/FunctionRetrieveDefaultDefinitionParameters"
+            },
+            "description": "Parameters used to specify the type of function to retrieve the default definition for."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/StreamingJobNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FunctionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the function's default definition.",
+            "schema": {
+              "$ref": "#/definitions/Function"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Function": {
```

Comment:
@veronicagg If you are referring to deep PUT, that would be in streamingjobs.json. Streaming jobs are our top-level resource with inputs, outputs, transformation, and functions being nested-resources under it.